sdks/go/pkg/beam/core/runtime/options.go

```diff
@@ -65,16 +65,22 @@ type RawOptions struct {
 // TODO(herohde) 3/12/2018: remove the extra options wrapper and the bogus
 // fields current required by the Java runners.
 
+// AdditionalOptions contains options in addition to those listed in RawOptions
+type AdditionalOptions struct {
+	Experiments []string `json:"experiments"`
+}
+
 // RawOptionsWrapper wraps RawOptions to the form expected by the
 // harness. The extra layer is currently needed due to Dataflow
 // expectations about this representation. Subject to change.
 type RawOptionsWrapper struct {
-	Options      RawOptions `json:"beam:option:go_options:v1"`
-	Runner       string     `json:"beam:option:runner:v1"`
-	AppName      string     `json:"beam:option:app_name:v1"`
-	Experiments  []string   `json:"beam:option:experiments:v1"`
-	RetainDocker bool       `json:"beam:option:retain_docker_containers:v1"`
-	Parallelism  int        `json:"beam:option:parallelism:v1"`
+	Options           RawOptions        `json:"beam:option:go_options:v1"`
+	Runner            string            `json:"beam:option:runner:v1"`
+	AppName           string            `json:"beam:option:app_name:v1"`
+	Experiments       []string          `json:"beam:option:experiments:v1"`
+	RetainDocker      bool              `json:"beam:option:retain_docker_containers:v1"`
+	Parallelism       int               `json:"beam:option:parallelism:v1"`
+	AdditionalOptions AdditionalOptions `json:"options"`
```

Comment:
The complaint the Python runner has is this field. So, we'll have to figure out alternative plumbing for the experiment signal, which likely means either duplicating it into one of the existing fields manually, OR having a different struct used at decoding time to be able to extract the existing value.

Reply:
Thanks for pointing it out. I've updated the code to populate experiments inside beam:option:go_options:v1. 
I also tried to put experiments inside beam:option:experiments:v1 but unfortunately it's not allowed by java runners so I guess moving it to beam:option:go_options:v1 seems to be the only option now